### PR TITLE
Move AsyncCallback to its own Class

### PR DESCRIPTION
**Pull request by alexw91**
*Issue #, if available:* N/A

*Description of changes:*
Move `AsyncCallback` into it's own class in preparation of adding `aws-c-http` integrations.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/native/mqtt_connection.c [hunk 1]
```diff
@@ -42,7 +42,7 @@ static struct {
 } s_async_callback = {0};
 
 void s_cache_async_callback(JNIEnv *env) {
-    jclass cls = (*env)->FindClass(env, "software/amazon/awssdk/crt/mqtt/MqttConnection$AsyncCallback");
+    jclass cls = (*env)->FindClass(env, "software/amazon/awssdk/crt/AsyncCallback");
```
Comment (justinboswell) [inline src/native/mqtt_connection.c]: The async header struct and this function should probably move into crt.c or something, and have a header which externs it, so you can use it in http and wherever else we need callbacks.

Comment (alexw91) [reply src/native/mqtt_connection.c]: Fixed

File: src/native/async_callback.c [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <assert.h>
+#include <jni.h>
+
+#include "async_callback.h"
+
+struct crt_async_callback global_async_callback;
```
Comment (justinboswell) [inline src/native/async_callback.c]: g_ is sufficient, and matches our coding standard

Comment (alexw91) [reply src/native/async_callback.c]: Fixed

**Pull request merged**